src/connector/src/sink/postgres.rs

```diff
@@ -0,0 +1,625 @@
+                        }
+                    }
+                }
+            }
+        } else {
+            for chunk in self.buffer.drain() {
+                for (op, row) in chunk.rows() {
+                    match op {
+                        Op::Insert => {
+                            self.client
+                                .execute_raw(&self.insert_statement, row.iter())
+                                .await?;
+                        }
+                        Op::UpdateInsert => {
+                            // NOTE(kwannoel): Here we use `MERGE` rather than `UPDATE/INSERT` directly.
+                            // This is because the downstream db could have cleaned the old record,
+                            // in that case it needs to be `INSERTED` rather than UPDATED.
+                            // On the other hand, if the record is there, it should be `UPDATED`.
+                            self.client
+                                .execute_raw(self.merge_statement.as_ref().unwrap(), row.iter())
+                                .await?;
+                        }
+                        Op::Delete => {
+                            self.client
+                                .execute_raw(
+                                    self.delete_statement.as_ref().unwrap(),
+                                    row.project(&self.pk_indices).iter(),
+                                )
+                                .await?;
+                        }
+                        Op::UpdateDelete => {}
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[async_trait]
+impl SinkWriter for PostgresSinkWriter {
+    async fn begin_epoch(&mut self, _epoch: u64) -> Result<()> {
+        Ok(())
+    }
+
+    async fn write_batch(&mut self, chunk: StreamChunk) -> Result<()> {
+        let cardinality = self.buffer.push(chunk);
+        if cardinality >= self.config.max_batch_rows {
+            self.flush().await?;
+        }
+        Ok(())
+    }
+
+    async fn barrier(&mut self, is_checkpoint: bool) -> Result<Self::CommitMetadata> {
+        if is_checkpoint {
+            self.flush().await?;
+        }
+        Ok(())
+    }
+
+    async fn abort(&mut self) -> Result<()> {
+        Ok(())
+    }
+
+    async fn update_vnode_bitmap(&mut self, _vnode_bitmap: Arc<Bitmap>) -> Result<()> {
+        Ok(())
+    }
+}
+
+fn data_type_not_supported(data_type_name: &str) -> SinkError {
+    SinkError::Postgres(anyhow!(format!(
+        "{data_type_name} is not supported in SQL Server"
+    )))
+}
+
+fn check_data_type_compatibility(data_type: DataType) -> Result<()> {
+    match data_type {
+        DataType::Boolean
+        | DataType::Int16
+        | DataType::Int32
+        | DataType::Int64
+        | DataType::Float32
+        | DataType::Float64
+        | DataType::Decimal
+        | DataType::Date
+        | DataType::Varchar
```

Comment:
RW `varchar` may need to sink to downstream `uuid` type, since we don't support `uuid` natively, so we have a casting logic in the jdbc sink.